components/proxy/src/main/java/com/hotels/styx/proxy/backends/file/FileChangeMonitor.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.proxy.backends.file;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.hash.HashCode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.hotels.styx.common.Files.fileContentMd5;
+import static java.lang.String.format;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.getLastModifiedTime;
+import static java.nio.file.Files.isReadable;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Monitors a file system object and notifies the consumer of any changes.
+ */
+public class FileChangeMonitor implements FileMonitor {
+    private final Path monitoredFile;
+    private final ScheduledExecutorService executor = newSingleThreadScheduledExecutor();
+    private final long pollPeriod;
+    private final TimeUnit timeUnit;
+    private ScheduledFuture<?> monitoredTask;
+
+    private AtomicReference<FileTime> lastChangedTime = new AtomicReference<>(FileTime.fromMillis(0));
+    private AtomicReference<HashCode> hashCode = new AtomicReference<>();
+    private volatile boolean performHashCheck;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileChangeMonitor.class);
+
+    @VisibleForTesting
+    FileChangeMonitor(String monitoredFile, long pollPeriod, TimeUnit timeUnit) {
+        requireExists(requireNonNull(monitoredFile));
+        this.monitoredFile = Paths.get(monitoredFile);
+        this.pollPeriod = pollPeriod;
```

Comment:
We can encapsulate some of these fields into our `Schedule` class if we move it out of `com.hotels.styx.client.healthcheck`, maybe into `common`? Just an idea.

Reply:
Hmm... we can think about this. I'll make a mental note of it.